### PR TITLE
Persist chat config

### PR DESCRIPTION
**Pull request by AbdBarho**
Closes https://github.com/LAION-AI/Open-Assistant/issues/2224

### CODE REVIEW

File: website/src/components/Chat/ChatSection.tsx [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
+          ...modelInfos[0].parameter_configs[0].sampling_parameters,
+          model_config_name: modelInfos[0].name,
+        },
   });
 
+  const config = form.watch();
+
+  useEffect(() => {
+    localStorage.setItem(CHAT_CONFIG_KEY, JSON.stringify(config));
+  }, [config]);
```
Comment (notmd) [inline website/src/components/Chat/ChatSection.tsx:31]: can use extract this to an empty component? watching on root component causes a lot rerender. You result the form context, so no need to pass any props.

File: website/src/components/Chat/ChatSection.tsx [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
 import { ChatConfigSummary } from "./ChatConfigSummary";
 import { useChatContext } from "./ChatContext";
 import { ChatConversation } from "./ChatConversation";
 
+const CHAT_CONFIG_KEY = "CHAT_CONFIG";
+
 export const ChatSection = ({ chatId }: { chatId: string }) => {
   const { modelInfos } = useChatContext();
 
+  const oldConfig = localStorage.getItem(CHAT_CONFIG_KEY);
```
Comment (notmd) [inline website/src/components/Chat/ChatSection.tsx:14]: Isn't this crash on the server side?

File: website/src/components/Chat/ChatSection.tsx (hunk 1 above)
Comment (AbdBarho) [reply website/src/components/Chat/ChatSection.tsx:31]: The `ChatSection` re-renders every time we change anything on the form, so this hook won't render any more than usual.

Anyway, I moved it to its own hook.

File: website/src/components/Chat/ChatSection.tsx (hunk 2 above)
Comment (AbdBarho) [reply website/src/components/Chat/ChatSection.tsx]: It is weird why it did not crash on my side or in the CI pipeline?

I added a check for it

**Review by notmd — APPROVED** [review]
The rerender when updating config is caused by the form context. I will refactor this when I work on new chat layout.

**Pull request merged**